test/models/user_test.rb

```diff
@@ -46,6 +46,14 @@ def setup
     assert_equal -74.0059731, u.longitude
   end
 
+  test 'accepts non-us zipcodes (UK)' do
+    u = build(:user, latitude: nil, longitude: nil)
+
+    u.update_attributes(zip: 'HP2 4HG')
+    assert_equal 40.7143528, u.latitude
+    assert_equal -74.0059731, u.longitude
+  end
+
```

Comment:
Can you add a test for the negative case, where the new validation will not be satisfied, and the errors will be added

Reply:
So I fixed this this morning and it led me down a little rabbit hole of making all API requests mocked with VCR. This also involved replacing/removing some stubbing done with Geocoder and setting up a little Monkey patch I found. It creates a new cassette for every test automatically and names it the test name. It made all the stubs in user unnecessary save for one which I could remake using mocha.